money-core/src/main/scala/com/comcast/money/core/CoreSpan.scala

```diff
@@ -32,65 +43,139 @@ import scala.collection.concurrent.TrieMap
  */
 case class CoreSpan(
   id: SpanId,
-  name: String,
+  var name: String,
```

Comment:
This would be a breaking change for any consumer directly using `CoreSpan` instead of going through `SpanFactory`.  I don't think it's avoidable, though, and these classes should probably be considered internal.

Reply:
If we are making this mutable I would personally prefer to then make the CoreSpan package private

Reply:
Seems like all of the `Core*` classes can be made package private without breaking the other projects with the exception of `CoreSpanInfo` which is used by a couple of other projects directly (e.g. `money-wire` and `money-spring`).  As `CoreSpanInfo` is really just a data carrier I wonder if it's worth moving the implementation to `money-api` with a factory method defined on `SpanInfo` in order to create it.

Reply:
Hmm, seems like a leak if other projects rely on `CoreSpanInfo` instead of `SpanInfo` interface.  Perhaps I got lazy?